pkg/daemon/update.go

```diff
@@ -1008,7 +1043,9 @@ func (dn *Daemon) writeUnits(units []igntypes.Unit) error {
 
 		if u.Contents != "" {
 			glog.Infof("Writing systemd unit %q", u.Name)
-
+			if err := createOrigFile(fpath); err != nil {
```

Comment:
Hm, there is a regressive behaviour here:

Let's say for example, we have a systemd unit deployed to the system via a MCO that had no backups attached to it (call it MC1). At some point we upgrade to this PR, and then write a new systemd unit file to overwrite it (call it MC2).

When this createOrigFile is called, it sees that:
1. this file is present on disk
2. there is no orig/no noorig associated with it

It then believes that the MC1 created unit belongs to the originally shipped MCO and creates an orig file. and thus if we were ever to delete both MCs it would "restore" a systemd unit that it thinks was on disk but it wasn't, much like what happened with etcd for files.

I think we have to harden it by checking whether the file existed on disk prior to restore, like the fix we had for the files case.

Reply:
Dang, right I think since units and dropins are under /etc we only need one check from the preiovus fix ti files. I’ll fix this thanks for catching it!

Reply:
alrighty, so I've modified the code to instead create the orig for units/dropins _only_ if the file is in /usr/etc instead of the logic we have in delete which is definitely more complex. It should serve the same tho as we won't create the orig for units/dropins if the file isn't shipped by rhcos/fcos

ptal @yuqi-zhang